hedgehog/src/Hedgehog/Internal/Predicate.hs

```diff
@@ -0,0 +1,12 @@
+module Hedgehog.Internal.Predicate
+  ( fromPred
+  ) where
+
+import Control.Monad (guard)
+
+-- | Convert a Boolean predicate to something suitable for use
+-- with @mapMaybe@-like functions. The @wizards@ package calls this
+-- @ensure@. @protolude@ and @relude@ generalize from 'Maybe' to
+-- arbitrary 'Control.Applicative.Alternative' and call it @guarded@.
```

Comment:
(Extreme nitpick) - need to push those comments on line (8-10) two space deeper, then add an empty `--` line, if we want this to be consistent with all the other commented stuff, like so:

```
-- | Convert a Boolean predicate to something suitable for use
--   with @mapMaybe@-like functions. The @wizards@ package calls this
--   @ensure@. @protolude@ and @relude@ generalize from 'Maybe' to
--   arbitrary 'Control.Applicative.Alternative' and call it @guarded@.
--
```

(Absolutely *not* a blocking issue, of course!) :incoming_envelope:

Reply:
Not sure if we even want a dedicated module for this, as we can just declare this in both `Gen` and `Tree` (but not expose it). Then, if we need to use it on a *third* module, we can consider [the rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) and move it to its own module.

It's just that Hedgehog.Internal.Predicate looks like a utility module that we don't really need (yet)...